samples/point_sprites.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <filament/Engine.h>
+#include <filament/IndexBuffer.h>
+#include <filament/Material.h>
+#include <filament/MaterialInstance.h>
+#include <filament/RenderableManager.h>
+#include <filament/Scene.h>
+#include <filament/TransformManager.h>
+#include <filament/VertexBuffer.h>
+#include <filament/View.h>
+
+#include <image/ImageSampler.h>
+#include <image/LinearImage.h>
+
+#include "../samples/app/Config.h"
+#include "../samples/app/FilamentApp.h"
+
+#include <cmath>
+
+#include "generated/resources/resources.h"
+
+using namespace filament;
+
+using utils::Entity;
+using utils::EntityManager;
+
+using MinFilter = TextureSampler::MinFilter;
+using MagFilter = TextureSampler::MagFilter;
+
+struct App {
+    VertexBuffer* vb;
+    IndexBuffer* ib;
+    Material* mat;
+    MaterialInstance* matInstance;
+    Camera* cam;
+    Texture* tex;
+    Entity renderable;
+};
+
+struct Vertex {
+    filament::math::float2 position;
+    uint32_t color;
+};
+
+#define NUM_POINTS 100
+#define TEXTURE_SIZE 128
+#define MAX_POINT_SIZE 128.0f
+
+void createSplatTexture(App& app, Engine* engine) {
+    static image::LinearImage splat(3, 3, 1);
+    splat.getPixelRef(1, 1)[0] = 1.0f;
+    splat = image::resampleImage(splat, TEXTURE_SIZE, TEXTURE_SIZE, image::Filter::GAUSSIAN_SCALARS);
+
+    Texture::PixelBufferDescriptor buffer(splat.getPixelRef(),
+            size_t(TEXTURE_SIZE * TEXTURE_SIZE * 4), Texture::Format::R, Texture::Type::FLOAT);
```

Comment:
Maybe use `sizeof(float)` here instead of 4 to help readability?

Reply:
Thanks, fixed.